lib/ansible/modules/storage/purestorage/purefa_info.py

```diff
@@ -0,0 +1,949 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2019, Simon Dodsley (simon@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: purefa_info
+version_added: '2.9'
+short_description: Collect information from Pure Storage FlashArray
+description:
+  - Collect information from a Pure Storage Flasharray running the
+    Purity//FA operating system. By default, the module will collect basic
+    information including hosts, host groups, protection
+    groups and volume counts. Additional information can be collected
+    based on the configured set of arguements.
+author:
+  - Pure Storage ansible Team (@sdodsley) <pure-ansible-team@purestorage.com>
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will define the information to be collected.
+        Possible values for this include all, minimum, config, performance,
+        capacity, network, subnet, interfaces, hgroups, pgroups, hosts,
+        admins, volumes, snapshots, pods, vgroups, offload, apps and arrays.
+    type: list
+    required: false
+    default: minimum
+extends_documentation_fragment:
+  - purestorage.fa
+'''
+
+EXAMPLES = r'''
+- name: collect default set of information
+  purefa_info:
+    fa_url: 10.10.10.2
+    api_token: e31060a7-21fc-e277-6240-25983c6c4592
+
+- name: collect configuration and capacity information
+  purefa_info:
+    gather_subset:
+      - config
+      - capacity
+    fa_url: 10.10.10.2
+    api_token: e31060a7-21fc-e277-6240-25983c6c4592
+
+- name: collect all information
+  purefa_info:
+    gather_subset:
+      - all
+    fa_url: 10.10.10.2
+    api_token: e31060a7-21fc-e277-6240-25983c6c4592
```

Comment:
Maybe add `register: result` to one of the examples and then a `debug` task to view the result (or parts of it). That makes it easier for not so experienced people to use the module.